catkin_tools/verbs/catkin_build/job.py

```diff
@@ -107,24 +107,38 @@ def next(self):
 
 
 def create_env_file(package, context):
+    # Exporting _CATKIN_SETUP_DIR supports DESTDIR, where the absolute location of the installspace
+    # at build time does not match the intended final location of it.
+    source_snippet = '_CATKIN_SETUP_DIR={space_path} . "{source_path}"'
     sources = []
-    source_snippet = '. "{source_path}"'
     # If installing to isolated folders or not installing, but devel spaces are not merged
     if (context.install and context.isolate_install) or (not context.install and context.isolate_devel):
         # Source each package's install or devel space
         space = context.install_space_abs if context.install else context.devel_space_abs
-        # Get the recursive dependcies
+        # Get the recursive dependencies
         depends = get_cached_recursive_build_depends_in_workspace(package, context.packages)
         # For each dep add a line to source its setup file
         for dep_pth, dep in depends:
-            source_path = os.path.join(space, dep.name, 'setup.sh')
-            sources.append(source_snippet.format(source_path=source_path))
+            space_path = os.path.join(space, dep.name)
+            source_path = os.path.join(space_path, 'setup.sh')
+            sources.append(source_snippet.format(space_path=space_path, source_path=source_path))
     else:
         # Just source common install or devel space
-        source_path = os.path.join(
-            context.install_space_abs if context.install else context.devel_space_abs,
-            'setup.sh')
-        sources = [source_snippet.format(source_path=source_path)] if os.path.exists(source_path) else []
+        if context.install:
+            if context.destdir:
+                # Intentionally not using os.path.join in this instance, as it's expected that
+                # when DESTDIR is specified, the install space is an absolute path, so a straight
+                # concatenation is more suitable than os.path.join.
+                space_path = context.destdir + context.install_space_abs
```

Comment:
Yeah, that was the reason, and you're right it should say so. I just didn't want it to become a novel. I can try again, or switch it to [do what catkin does](https://github.com/ros/catkin/commit/7282a4bb832610e16d1a63c8336be41d781cb0e6#diff-616a0c8f980657f6b84468d0ce4abc77R377). What's your preference?